cmsl1t/producers/jets.py

```diff
@@ -0,0 +1,93 @@
+from importlib import import_module
+
+DEFAULT_ATTRIBUTES = [
+    'etCorr', 'muMult', 'eta', 'phi', 'nhef', 'pef', 'mef', 'chMult',
+    'elMult', 'nhMult', 'phMult', 'chef', 'eef'
+]
+
+L1T_Jet_ATTRIBUTES = [
+    'jetEt', 'jetPhi', 'jetEta',
+]
+
+
+class Jet(object):
+    '''
+        Create a simple python wrapper for
+        L1Analysis::L1AnalysisRecoJetDataFormat
+    '''
+
+    def __init__(self, event, index, prefix='Jet_'):
+        global DEFAULT_ATTRIBUTES
+        # this could be simplified with a list of attributes
+        read_attributes = []
+        if not hasattr(event, prefix + DEFAULT_ATTRIBUTES[0]):
+            read_attributes[:] = L1T_Jet_ATTRIBUTES[:]
+            self.sortBy = 'jetEt'
+        else:
+            read_attributes[:] = DEFAULT_ATTRIBUTES[:]
```

Comment:
What's the advantage of slicing here?

Reply:
It is the overwriting existing object vs allocating a new one (see Tai's event loops)